project-templates/node-vscode-json/glsp-server/src/diagram/tasklist-diagram-module.ts

```diff
@@ -0,0 +1,91 @@
+/********************************************************************************
+ * Copyright (c) 2022 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import {
+    ActionHandlerConstructor,
+    ComputedBoundsActionHandler,
+    DiagramConfiguration,
+    DiagramModule,
+    GModelFactory,
+    GModelIndex,
+    InstanceMultiBinding,
+    LabelEditValidator,
+    ModelState,
+    OperationHandlerConstructor,
+    SourceModelStorage
+} from '@eclipse-glsp/server-node';
+import { BindingTarget } from '@eclipse-glsp/server-node/lib/di/binding-target';
+import { injectable, interfaces } from 'inversify';
+import { CreateTaskHandler } from '../handler/create-task-node-handler';
+import { TasklistApplyLabelEditHandler } from '../handler/tasklist-apply-label-edit-handler';
+import { TasklistChangeBoundsHandler } from '../handler/tasklist-change-bounds-handler';
+import { TasklistDeleteHandler } from '../handler/tasklist-delete-handler';
+import { TasklistLabelEditValidator } from '../handler/tasklist-label-edit-validator';
+import { TasklistGModelFactory } from '../model/tasklist-gmodel-factory';
+import { TasklistModelIndex } from '../model/tasklist-model-index';
+import { TasklistModelState } from '../model/tasklist-model-state';
+import { TasklistStorage } from '../model/tasklist-storage';
+import { TasklistDiagramConfiguration } from './tasklist-diagram-configuration';
+
+@injectable()
+export class TasklistDiagramModule extends DiagramModule {
+    readonly diagramType = 'minimal-diagram';
+
+    protected override configure(
+        bind: interfaces.Bind,
+        unbind: interfaces.Unbind,
+        isBound: interfaces.IsBound,
+        rebind: interfaces.Rebind
+    ): void {
+        bind(TasklistModelIndex).toSelf().inSingletonScope();
```

Comment:
Why is this binding before the super-call? This looks surprising. Is this required to ensure that when `bindGModelIndex()` is called, this already exists?

This isn't a blocking issue, but I think it'd be much nicer if that all could be handled in `bindGModelIndex()`.
For instance, wouldn't it work if all those methods below would get `bind`, `unbind`, etc. provided as parameter? This way, they could at least bind their services in the same method before they return the class to be bound as service?

Reply:
I aggree that is workaround is not nice and I already have openend a GLSP issue for that (https://github.com/eclipse-glsp/glsp/issues/631). 
For now we can get around this by using the binding context that is provided by the super implementation. I will update this accordingly